youtube_transcript_api/formatters.py

```diff
@@ -0,0 +1,174 @@
+                    line['text']for line in transcript_data))
+
+
+class SRTTranscriptFormatter(TranscriptFormatter):
+    """Formatter for outputting the SRT Format
+
+    Converts the fetched transcript data into a simple .srt file format.
+    """
+    DELIMITER = '\n\n'
+
+    @classmethod
+    def format(cls, transcript_data):
+        output = []
+        for frame, item in enumerate(transcript_data, start=1):
+            start_time = float(item.get('start'))
+            duration = float(item.get('duration', '0.0'))
+
+            output.append("{frame}\n".format(frame=frame))
+            output.append("{start_time} --> {end_time}\n".format(
+                start_time=parse_timecode(start_time),
+                end_time=parse_timecode(start_time + duration)
+            ))
+            output.append("{text}".format(text=item.get('text')))
+            if frame < len(transcript_data):
+                output.append('\n\n')
+        return '{}\n'.format(''.join(output))
+
+
+class TranscriptFormatterFactory(object):
+    """A Transcript Class Factory
+
+    Allows for adding additional custom Transcript classes for the API
+    to use. Custom Transcript classes must inherit from the
+    TranscriptFormatter abstract base class.
+    """
+    def __init__(self):
+        self._formatters = defaultdict(JSONTranscriptFormatter)
+
+    def add_formatter(self, name, formatter_class):
+        """Allows for creating additional transcript formatters.
+
+
+        :param name: a name given to the `formatter_class`
+        :type name: str
+        :param formatter_class: a subclass of TranscriptFormatter
+        :type formatter_class: class
+        :rtype None
+        """
+        if not issubclass(formatter_class, TranscriptFormatter):
+            raise TypeError((
+                '{0} must be a subclass of TranscriptFormatter'
+                ).format(formatter_class)
+            )
+        self._formatters.update({name: formatter_class})
+
+    def add_formatters(self, formatters_dict):
+        """Allow creation of multiple transcript formatters at a time.
+
+        :param formatters_dict: key(s) are the string name to be given
+         to the formatter class, value for each key should be a subclass
+         of TranscriptFormatter.
+        :type formatters_dict: dict
+        :rtype None
+        """
+        for name, formatter_class in formatters_dict.items():
+            self.add_formatter(name, formatter_class)
+
+    def get_formatter(self, name):
+        """Retrieve a formatter class by its assigned name.
+
+        :param name: the string name given to the formatter class.
+        :type name: str
+        :return: a subclass of `TranscriptFormatter`
+        """
+        return self._formatters[name]
+
+
+formats = TranscriptFormatterFactory()
+formats.add_formatters({
+    'json': JSONTranscriptFormatter,
+    'srt': SRTTranscriptFormatter,
+    'text': TextTranscriptFormatter
+})
```

Comment:
I wouldn't want this feature to change the default behaviour of this module in case no format is provided by the user. Therefore, I'd suggest also adding a `PrettyPrintFormatter` which implements the current default behaviour.

Reply:
The current behavior for this is hidden away behind that `defaultdict` 

```python
# because -> self._formatters = defaultdict(JSONTranscriptFormatter)

def get_formatter(self, name):
        return self._formatters[name]  # if name is None, JSONTranscriptFormatter should be returned.
```

I could totally make this default instead to `PrettyPrintFormatter` if that is preferred instead. So then do you still want the default for CLI to be json? If so, then it would make sense to change the default `parsed_args.format` to no longer be `None` and instead be `'json'` and in a custom script a user would then have to specify `format='json'` explicitly.

Reply:
As commented above, the current default formatting is the pprint, not json.